plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TableSnapshot.java

```diff
@@ -174,30 +176,32 @@ public Stream<DeltaLakeTransactionLogEntry> getCheckpointTransactionLogEntries(
         LastCheckpoint checkpoint = lastCheckpoint.get();
         // Add entries contain statistics. When struct statistics are used the format of the Parquet file depends on the schema. It is important to use the schema at the time
         // of the Checkpoint creation, in case the schema has evolved since it was written.
-        Optional<MetadataEntry> metadataEntry = entryTypes.contains(ADD) ?
-                Optional.of(getCheckpointMetadataEntry(
-                        session,
-                        checkpointSchemaManager,
-                        typeManager,
-                        fileSystem,
-                        stats,
-                        checkpoint)) :
-                Optional.empty();
+        Optional<MetadataAndProtocolEntry> metadataAndProtocol = Optional.empty();
+        if (entryTypes.contains(ADD)) {
+            metadataAndProtocol = Optional.of(getCheckpointMetadataEntry(
```

Comment:
pre-existing: potential area of improvement we call `getCheckpointTransactionLogEntries` 2 times when dealing with `ADD` - once for metadata & protocol and once for the `entryTypes`. With some extra logic on `CheckpointEntryIterator` we could potentially spare the first checkpoint file access